src/SipHash.chpl

```diff
@@ -99,11 +106,12 @@ module SipHash {
     var v1 = 0x646f72616e646f6d: uint(64);
     var v2 = 0x6c7967656e657261: uint(64);
     var v3 = 0x7465646279746573: uint(64);
-    const k0 = 0xf0e1d2c3b4a59687: uint(64);
-    const k1 = 0x79695a4b3c2d1e0f: uint(64);
+    const k0 = 0x0706050403020100: uint(64);
+    const k1 = 0x0f0e0d0c0b0a0908: uint(64);
     var m: uint(64);
     var i: int;
-    const lastPos = D.low + D.size - (D.size % 8);
+    const lastPos = if isSubtype(msg.type, c_ptr) then D.size - (D.size % 8)
+      else D.low + D.size - (D.size % 8); // if C index, 0-up
```

Comment:
I think you can drop this change since we 0-up at the call site by passing in `D=0..#l`

Reply:
ah, cool. I will just pull in your changes, i think, since you have already tested them.